src/solver/expressions/visitors/AstDOTStyleVisitor.cpp

```diff
@@ -0,0 +1,254 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+
+#include "antares/solver/expressions/visitors/AstDOTStyleVisitor.h"
+
+#include <algorithm>
+#include <set>
+
+#include <antares/solver/expressions/nodes/ExpressionsNodes.h>
+
+namespace Antares::Solver::Visitors
+{
+namespace NodeStyle
+{
+static constexpr BoxStyle SumStyle{"aqua", "hexagon", "filled, solid"};
+static constexpr BoxStyle BinaryStyle{"aqua", "oval", "filled, rounded"};
+static constexpr BoxStyle ComparisonStyle{"yellow", "diamond", "filled"};
+static constexpr BoxStyle NegationStyle{"tomato", "invtriangle", "filled, solid"};
+static constexpr BoxStyle LiteralStyle{"lightgray", "box", "filled, solid"};
+static constexpr BoxStyle VariableStyle{"gold", "box", "filled, solid"};
+static constexpr BoxStyle ParameterStyle{"wheat", "box", "filled, solid"};
+static constexpr BoxStyle ComponentParameterStyle{"springgreen", "octagon", "filled, solid"};
+static constexpr BoxStyle ComponentVariableStyle{"goldenrod", "octagon", "filled, solid"};
+static constexpr BoxStyle PortFieldStyle{"olive", "component", "filled, solid"};
+} // namespace NodeStyle
+
+void GetLegend(const std::map<std::string, std::map<const Nodes::Node*, unsigned int>>& nodeIds,
+               std::ostream& os)
+{
+    os << R"raw(subgraph cluster_legend {
+label = "Legend";
+style = dashed;
+fontsize = 16;
+color = lightgrey;
+node [shape=plaintext];
+
+)raw";
+
+    auto order = nodeIds.size() <=> 1;
+    if (order == std::strong_ordering::greater)
```

Comment:
The spaceship operator is nice, but it makes the code less readable IMO.

```suggestion
    if (nodeIds.size() > 1)
```